crates/sui-types/src/object.rs

```diff
@@ -922,14 +922,18 @@ impl Object {
         Self::immutable_with_id_for_testing(IMMUTABLE_OBJECT_ID.with(|id| *id))
     }
 
-    /// make a test shared object.
+    /// Make a test shared object. Note that this function returns the same object called from the same thread.
     pub fn shared_for_testing() -> Object {
         thread_local! {
             static SHARED_OBJECT_ID: ObjectID = ObjectID::random();
```

Comment:
I am actually curious why this is a static local. Not very intuitive.
But that's a separate story and shouldn't block you from merging this PR

Reply:
Let me try to dig into it.